sklearn/neural_network/_rbm.py

```diff
@@ -370,7 +370,12 @@ def score_samples(self, X):
         ind = (np.arange(v.shape[0]), rng.randint(0, v.shape[1], v.shape[0]))
         if sp.issparse(v):
             data = -2 * v[ind] + 1
-            v_ = v + sp.csr_matrix((data.A.ravel(), ind), shape=v.shape)
+            if sp.isspmatrix_csr(v):
+                # `data` is a NumPy matrix with the previous indexing
+                v_ = v + sp.csr_matrix((data.A.ravel(), ind), shape=v.shape)
+            else:
+                # `data` is a NumPy array with the previous indexing
+                v_ = v + sp.csr_matrix((data.ravel(), ind), shape=v.shape)
```

Comment:
OK `sp.isspmatrix` did not make the difference between arrays and matrices for SciPy < 1.10. Let's go back to your previous solution with additinal comment


```suggestion
            if isinstance(data, np.matrix):  # v is a sparse matrix
                v_ = v + sp.csr_matrix((data.A.ravel(), ind), shape=v.shape)
            else:  # v is a sparse array
                v_ = v + sp.csr_array((data.ravel(), ind), shape=v.shape)
```